clients/admin-ui/src/features/common/system-data-flow/DataFlowAccordionForm.tsx

```diff
@@ -0,0 +1,221 @@
+import {
+  AccordionButton,
+  AccordionIcon,
+  AccordionItem,
+  AccordionPanel,
+  Button,
+  ButtonGroup,
+  Flex,
+  Spacer,
+  Stack,
+  Tag,
+  Text,
+  useDisclosure,
+  useToast,
+} from "@fidesui/react";
+import { FormGuard } from "common/hooks/useIsAnyFormDirty";
+import { GearLightIcon } from "common/Icon";
+import QuestionTooltip from "common/QuestionTooltip";
+import { DataFlowSystemsDeleteTable } from "common/system-data-flow/DataFlowSystemsDeleteTable";
+import DataFlowSystemsModal from "common/system-data-flow/DataFlowSystemsModal";
+import { errorToastParams,successToastParams } from "common/toast";
+import { Form, Formik, FormikBag } from "formik";
+import React, { useEffect, useMemo, useState } from "react";
+
+import { useAppSelector } from "~/app/hooks";
+import {
+  useGetAllSystemsQuery,
+  useUpdateSystemMutation,
+} from "~/features/system";
+import { selectAllSystems } from "~/features/system/system.slice";
+import { DataFlow, System } from "~/types/api";
+
+const defaultInitialValues = {
+  dataFlowSystems: [] as DataFlow[],
+};
+
+export type FormValues = typeof defaultInitialValues;
+
+type DataFlowAccordionItemProps = {
+  isIngress?: boolean;
+  system: System;
+  isSystemTab?: boolean;
+};
+
+export const DataFlowAccordionForm = ({
+  system,
+  isIngress,
+  isSystemTab,
+}: DataFlowAccordionItemProps) => {
+  const toast = useToast();
+  const flowType = isIngress ? "Source" : "Destination";
+  const pluralFlowType = `${flowType}s`;
+  const dataFlowSystemsModal = useDisclosure();
+  const [updateSystemMutationTrigger, mutationResult] =
+    useUpdateSystemMutation();
+
+  useEffect(() => {
+    if (mutationResult.isLoading || mutationResult.isUninitialized) {
+      return;
+    }
+
+    if (mutationResult.isError) {
+      toast(errorToastParams("Failed to update data flows"));
+    }
+    if (mutationResult.isSuccess) {
+      toast(successToastParams(`${pluralFlowType} updated`));
```

Comment:
I had to add this janky `useEffect` to properly show the error toast. The reason was the  `handleSubmit` call has an older closure with the previous `mutationResult` values. updating to use a `useCallback` didn't worth. I looked at the docks for a better pattern and I didn't see anything that covers this scenario. I did see a mention of a global middleware that will automatically handle toasts. 

https://redux-toolkit.js.org/rtk-query/usage/error-handling#handling-errors-at-a-macro-level

It might be worth adding this in because it will remove a lot of boilerplate. The only issues I see if we need special copy for different api slices.

Reply:
oh weird. sorry I didn't realize it'd be a big lift to add the error toast, otherwise we could've punted!